src/Randomizer.SMZ3.Tracking/AutoTracking/GameService.cs

```diff
@@ -40,6 +40,39 @@ public GameService(Tracker tracker, IItemService itemService, IWorldService worl
             _trackerPlayerId = worldAccessor.Worlds.Count > 0 ? worldAccessor.Worlds.Count : 0;
         }
 
+        /// <summary>
+        /// Updates memory values so both SM and Z3 will cancel any pending MSU resumes and play
+        /// all tracks from the start until new resume points have been stored.
+        /// </summary>
+        /// <returns>False if canceling MSU resume is not currently possible</returns>
+        public bool TryCancelMsuResume()
+        {
+            if (!IsInGame())
+            {
+                return false;
+            }
+
+            // Zero out SM's NO_RESUME_AFTER variable
+            AutoTracker?.WriteToMemory(new EmulatorAction()
+            {
+                Type = EmulatorActionType.WriteBytes,
+                Domain = MemoryDomain.WRAM,
+                Address = 0x7E033A, // As declared in sm/msu.asm
+                WriteValues = new List<byte>() { 0, 0 }
+            });
+
+            // Zero out Z3's MSUResumeTime variable
+            AutoTracker?.WriteToMemory(new EmulatorAction()
```

Comment:
You should only do these for the particular game the player is in. According to https://jathys.zophar.net/supermetroid/kejardon/RAMMap.txt, this memory address in SM is in use:
7E:1E17 - 7E:1E66    Variable use PLM value

Reply:
Good catch. For some reason, I had it in my head that Z3 could resume even when transitioning from SM, but `0x7E1E6B` does indeed get zeroed out during the switch between games.

Reply:
Well, I think you're correct. I believe it does still restore when switching to SM and switching back. I think the memory values get saved, then restored later. It's probably something that should be resolved, but I think it'll have to be resolved separately.

Reply:
I gave it a try and the Z3 memory location zeroes out during the transition to SM and stays zero when transitioning back, so they must not be dumping that one to SRAM.